service/src/main/java/com/codeforcommunity/requester/S3Requester.java

```diff
@@ -369,4 +377,28 @@ public static void deleteHTML(String name) {
 
     externs.getS3Client().deleteObject(awsRequest);
   }
+
+  public static List<String> getAllNamesinBucket(String bucketName) {
+    ListObjectsV2Request req = new ListObjectsV2Request();
+    req.setBucketName(bucketName);
+    req.setDelimiter("/");
+    String path = String.join("/", TEMPLATE_S3_DIR, externs.getDirPublic());
+    req.setPrefix(path);
+    req.setStartAfter(path);
+
+    ListObjectsV2Result res;
+    try {
+
+      res = externs.getS3Client().listObjectsV2(req);
+      String prefix = req.getPrefix();
+      List<String> result;
+      result = res.getObjectSummaries().stream().map(s -> s.getKey().replace(prefix, "")).collect(Collectors.toList());
+      return result;
```

Comment:
⛏️ Can we also remove the template extension from the object names as well? This way, the frontend won't have to worry about removing the extensions, and it'll also match how we add and load templates - right now, those take just the template name (e.g. `test_email` rather than `test_email_template.html`)

Reply:
fixed and tested